akka-stream-testkit/src/main/scala/akka/stream/testkit/StreamTestKit.scala

```diff
@@ -309,6 +317,9 @@ object TestSubscriber {
       self
     }
 
+    def expectMsgPF[T](f: PartialFunction[AnyRef, T]): T =
+      f(probe.expectMsgType[AnyRef])
```

Comment:
Yep, if PF is meant to be over onNext. It can be over events though, so maybe rename this to expectEventPF() where the PartialFunciton is defined on PublisherEvent/SubscriberEvent instead of AnyRef. Then an expectNextPF() can be defined on top of that that does unwrapping properly.